clientapi/routing/profile.go

```diff
@@ -253,6 +295,38 @@ func SetDisplayName(
 	}
 }
 
+func getProfile(
```

Comment:
A docstring (with information about how this function will contact application services/other homeservers) would be handy.